.github/workflows/test-python.yml

```diff
@@ -25,10 +25,8 @@ jobs:
     name: CPython ${{ matrix.python-version }}-${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
-      - name: Setup Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: ${{ matrix.python-version }}
+      - name: Base Setup
+        uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
```

Comment:
This action takes care of installing python and setting up caching for pip and conda